### PR TITLE
WIP - refactor menu

### PR DESCRIPTION
**Pull request by tlsim**
@jleft I think I'm getting the idea after yesterday's chat. Am I on the right track (with the structure and the TODO items) ? 

Also removed sc.menu.group - it's a wrapper around array.map and selection.datum. Calling the generator components directly seems simpler.

### CODE REVIEW

**Comment by jleft** [comment]
:+1: Looks really good to me

File: src/assets/js/menu/side.js [hunk 1]
```diff
@@ -44,34 +44,41 @@
...
-            .generator(sc.menu.generator.toggleGroup())
+        var secondaryChartToggle = sc.menu.generator.toggleGroup()
             .on('optionChange', function(chart) {
                 dispatch.secondaryChartChange(chart);
             });
 
         var side = function(selection) {
-            selection.each(function() {
+            // TODO: var model = selection.datum();
+            selection.each(function() { // TODO: Do we really need to support multiple selection here ?
```
Comment (tlsim) [inline src/assets/js/menu/side.js]: Low-barrel example *does* use selection.each(), with model available as first argument to the function.

Comment (jleft) [inline src/assets/js/menu/side.js]: I'm not too fussed either way, but for consistency we could use `selection.each()` with all of our components?

I also think the latter of these examples (below) is a bit nicer:

```js
var model = selection.datum();
// ...
```

```js
selection.each(function(model) {
    // ...
});
```

Comment (tlsim) [inline src/assets/js/menu/side.js]: I agree. Let's use the second example and be consistent in using `selection.each()`.

Comment (jleft) [inline src/assets/js/menu/side.js]: #303

**Pull request merged**